py-rattler/rattler/networking/middleware.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import annotations
+from rattler.rattler import PyMirrorMiddleware, PyAuthenticationMiddleware
+
+
+class MirrorMiddleware:
+    def __init__(self, middlewares: dict[str, list[str]]) -> None:
+        self._middleware = PyMirrorMiddleware(middlewares)
```

Comment:
```suggestion
    def __init__(self, mirrors: dict[str, list[str]]) -> None:
        self._middleware = PyMirrorMiddleware(mirrors)
```
It would probably also make sense to docoument the format of `mirrors` here in a docstring.